lib/seven/otters/projection.ex

```diff
@@ -128,52 +130,29 @@ defmodule Seven.Otters.Projection do
         apply_events(events, new_state)
       end
 
-      defp rehydratate_by_snapshot(nil) do
+      defp rehydratate(true, last_event_id) do
         events =
-          unquote(listener_of_events)
-          |> Seven.EventStore.EventStore.events_by_types()
-          |> Seven.EventStore.EventStore.events_stream_to_list()
+          if last_event_id == nil do
+            unquote(listener_of_events)
+            |> Seven.EventStore.EventStore.events_by_types()
+          else
+            unquote(listener_of_events)
+            |> Seven.EventStore.EventStore.events_by_types(last_event_id)
+          end
+
+        events = events |> Seven.EventStore.EventStore.events_stream_to_list()
```

Comment:
? `Seven.EventStore.EventStore.events_stream_to_list()`
currently loads in memory all the events, if the event history is long this will be a memory problem.
should be stream.